src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/ViewportManagerWithPlatformFeatures.cs

```diff
@@ -582,11 +591,13 @@ void UpdateViewport(Rect viewport)
 				m_visibleWindow = currentVisibleWindow;
 			}
 
+#if !UNO_HAS_ENHANCED_LIFECYCLE
```

Comment:
@dr1rrb It caused some test failures, `ValidateNoScrollingSurfaceScenario` was one of them.

Reply:
@dr1rrb How does the performance impact manifests itself? @Youssef1313 if you remember which tests fail, it would be interesting to indicate which.

Reply:
> @dr1rrb How does the performance impact manifests itself? @Youssef1313 if you remember which tests fail, it would be interesting to indicate which.

By raising event more often than needed ... but we should have some tests for that. So if all EVP tests are passing (without changing/ignoring them of course :P) I would assume it's fine.